### DISCUSSION TITLE
Submit button alternative functionality.

### DISCUSSION THREAD

**Discussion by nikos445** [ansibleguy76/ansibleforms#415] [Q&A] [answered]
Feature Request: 
- Add extra functionality/options for submit button.

Options / Functionality:
- Trigger Webhooks instead of ansible playbook / awx / etc.

Example use case: 
- Trigger Jenkins Generic Webhook by executing POST HTTP request (http://example.com/test-url-path) and passing body in JSON format (extravars)

**Comment by nikos445** [comment]
I already implemented with Webhook in ansible playbook that is saved internally to persistent volume :) but i thought that it would be a great feature to add in the ansible forms. It will enhance its ability to trigger ansible playbooks in jenkins not only AWX

**Comment by nikos445** [comment]
Additionally, i found an issue that if you execute playbook locally, some extravars, are not passed. E.g. __jobid__

**Comment by ansibleguy76** [comment]
> Additionally, i found an issue that if you execute playbook locally, some extravars, are not passed. E.g. **jobid**

Can you make that a separate thread ?

**Comment by ansibleguy76** [comment]
> I already implemented with Webhook in ansible playbook that is saved internally to persistent volume :) but i thought that it would be a great feature to add in the ansible forms. It will enhance its ability to trigger ansible playbooks in jenkins not only AWX

i totally understand.  So instead of ansible and awx, a custom endpoint, perhaps using fnRestAdvanced.  Would not be a bad idea, let's reopen this.

**Comment by nikos445** [comment]
Please also consider the ability to **disable** submit button if required fields are not fulfilled (not being able to click it instead of evaluating the fields and showing error)

**Comment by ansibleguy76** [comment]
I will probably add the submit option to an expression.  Allow you to capture the extravars and do what you want.  REST post, store, ftp, ... many options would become possible.  Perhaps some sort of "plugin".  I'll have to thing this through.  let me know if you have ideas of your own

**Comment by nikos445** [comment, marked as answer]
For now I am using a local playbook (e.g webhook.yaml) that is using the "extravars" and a "URL" statically defined in the form after submittion. With the playbook I trigger the corresponding remote webhook passing a JSON (extravars) from the playbook. This is locally executed in Ansible Forms so Ansible Forms in reality is modular if you do it like that ;)
